pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -0,0 +1,92 @@
+package core
+
+import (
+	"testing"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/test/helper"
+)
+
+func Test_genericScheduler_AssignReplicas(t *testing.T) {
+	tests := []struct {
+		name       string
+		retryTimes int
```

Comment:
What is the reason for adding a retry here?

Reply:
it is the acceptance cases given by testers for this requirement.

she want to check when it meets random, In other word, retry multi times, the result should have both 2:1 and 1:2

Reply:
Thanks, I get it. 

This test case is a bit complex. Perhaps we can add some simple comments to explain the purpose of the test.

Reply:
How many times did you run the tests? 
My concern is that testing random by specific times might not stable, which means the test might fail, even though the probability is very slow.

Reply:
> This test case is a bit complex. Perhaps we can add some simple comments to explain the purpose of the test.

okey

Reply:
> How many times did you run the tests? My concern is that testing random by specific times might not stable, which means the test might fail, even though the probability is very slow.

10 times, in order to keep the same as `alpha test case` running by product tester.

the probability of fail is (1/2)^10= 0.1%